Scan reports "failed" state while tasks still running

## Specify type:
- Bug

### Priority:
- High
___

## Description:
Two things are happening here. 

One, the scan reports it is failed while tasks are still running. I think this is unclear and the scan should still report itself as "running" if there are tasks running.

Two, I was under the impression that the goal of breaking out the unreachable systems from the failed systems was to not report the scan as failed if there are unreachable systems.

## Bug Report


### Version Information:
Quipucords develoment server from master (b080c5176de390f690cbc71b71c8ce7227c93b99)
API interaction only

### Expected behavior:
If I run a scan that includes some unreachable systems, the scan will not report itself in some "stopped" state ('failed', 'completed', 'paused' or 'canceled') if there are tasks still running.

Additionally, I expect that unreachable systems do not cause a scan to be reported as "failed", rather, they are accounted as "uncreachable"

### Actual behavior:
Scan fails while tasks are still running.

Below is the test result that alerted me to this probem. It is expecting the sum of the systems_unreachable, systems_failed, and systems_scanned to be equal to "systems_count". 

```
E           AssertionError: Details of failed scan : {'connection_results': None,
E              'errors': [],
E              'expected_products': [{27: {'distribution': {}}},
E                                    {32: {'distribution': {'name': 'Red Hat Enterprise '
E                                                                   'Linux',
E                                                           'version': '7.4'},
E                                          'jbossas': {'install-process': 'zip',
E                                                      'version': '7.1.1 "Final"'},
E                                          'openjdk': {'version': '1.8'}}}],
E              'final_status': 'failed',
E              'inspection_results': [],
E              'report_id': None,
E              'scan_id': 2,
E              'scan_job_id': 2,
E              'scan_results': {'end_time': '2018-04-17T17:39:09.065048',
E                               'id': 2,
E                               'options': {'max_concurrency': 50},
E                               'scan': {'id': 2,
E                                        'name': 'e703a98c-e8fe-42e4-a222-c41f25db54cb'},
E                               'scan_type': 'inspect',
E                               'sources': [{'id': 27,
E                                            'name': 'ff310a61-85b6-4f83-87c7-64049c103f07',
E                                            'source_type': 'network'},
E                                           {'id': 32,
E                                            'name': '836ac9fe-e8b9-4e43-b02a-37fff1e11ce0',
E                                            'source_type': 'network'}],
E                               'start_time': '2018-04-17T17:38:57.484603',
E                               'status': 'failed',
E                               'status_message': 'One of more tasks did not complete.  See '
E                                                 'tasks for details.',
E                               'systems_count': 2,
E                               'systems_failed': 0,
E                               'systems_scanned': 0,
E                               'systems_unreachable': 1,
E                               'tasks': [{'end_time': '2018-04-17T17:39:08.373302',
E                                          'scan_type': 'connect',
E                                          'source': 27,
E                                          'start_time': '2018-04-17T17:38:57.507532',
E                                          'status': 'completed',
E                                          'status_message': 'Task is complete.',
E                                          'systems_count': 1,
E                                          'systems_failed': 0,
E                                          'systems_scanned': 0,
E                                          'systems_unreachable': 1},
E                                         {'end_time': '2018-04-17T17:39:09.041453',
E                                          'scan_type': 'connect',
E                                          'source': 32,
E                                          'start_time': '2018-04-17T17:39:08.378911',
E                                          'status': 'completed',
E                                          'status_message': 'Task is complete.',
E                                          'systems_count': 1,
E                                          'systems_failed': 0,
E                                          'systems_scanned': 1,
E                                          'systems_unreachable': 0},
E                                         {'end_time': '2018-04-17T17:39:09.059752',
E                                          'scan_type': 'inspect',
E                                          'source': 27,
E                                          'start_time': '2018-04-17T17:39:09.048929',
E                                          'status': 'failed',
E                                          'status_message': 'Scan task encountered error: '
E                                                            'Scan task failed.  Error: '
E                                                            'Inventory provided no '
E                                                            'reachable hosts.'},
E                                         {'scan_type': 'inspect',
E                                          'source': 32,
E                                          'start_time': '2018-04-17T17:39:09.075563',
E                                          'status': 'running',
E                                          'status_message': 'Task is running.',
E                                          'systems_count': 1,
E                                          'systems_failed': 0,
E                                          'systems_scanned': 0,
E                                          'systems_unreachable': 0}]},
E              'source_id_to_hostname': {27: 'bad-host', 32: 'sonar-jbossas-rhel-7-vc'},
E              'task_results': [{'end_time': '2018-04-17T17:39:08.373302',
E                                'scan_type': 'connect',
E                                'source': 27,
E                                'start_time': '2018-04-17T17:38:57.507532',
E                                'status': 'completed',
E                                'status_message': 'Task is complete.',
E                                'systems_count': 1,
E                                'systems_failed': 0,
E                                'systems_scanned': 0,
E                                'systems_unreachable': 1},
E                               {'end_time': '2018-04-17T17:39:09.041453',
E                                'scan_type': 'connect',
E                                'source': 32,
E                                'start_time': '2018-04-17T17:39:08.378911',
E                                'status': 'completed',
E                                'status_message': 'Task is complete.',
E                                'systems_count': 1,
E                                'systems_failed': 0,
E                                'systems_scanned': 1,
E                                'systems_unreachable': 0},
E                               {'end_time': '2018-04-17T17:39:09.059752',
E                                'scan_type': 'inspect',
E                                'source': 27,
E                                'start_time': '2018-04-17T17:39:09.048929',
E                                'status': 'failed',
E                                'status_message': 'Scan task encountered error: Scan task '
E                                                  'failed.  Error: Inventory provided no '
E                                                  'reachable hosts.'},
E                               {'scan_type': 'inspect',
E                                'source': 32,
E                                'start_time': '2018-04-17T17:39:09.075563',
E                                'status': 'running',
E                                'status_message': 'Task is running.',
E                                'systems_count': 1,
E                                'systems_failed': 0,
E                                'systems_scanned': 0,
E                                'systems_unreachable': 0}]}
E           assert 0 == ((1 - 0) - 0)

```

### Steps to reproduce:
Create a scan that includes one good network source/credential and one purposely unreachable network system (you should not be able to connect to it with the credential provided).

Inspect the results.

### Environment information:
Provide detailed information on the *scanning* and *target* systems (if applicable); see example below.

Type            | Operating System | OS Version | Python Version | Virtualization
--------------- | ---------------- | ---------- | -------------- | --------------
Scanning System | Fedora             | 26        | 3.6            | Bare Metal

--- a/quipucords/scanner/job.py
+++ b/quipucords/scanner/job.py
@@ -41,6 +41,7 @@ class ScanJobRunner(Process):
 
     def run(self):
         """Trigger thread execution."""
+        # pylint: disable=too-many-locals,too-many-statements
         # Job is not running so start
         self.scan_job.start()
         if self.scan_job.status != ScanTask.RUNNING:
@@ -73,6 +74,7 @@ class ScanJobRunner(Process):
         self.scan_job.log_message(
             'Queuing the following incomplete tasks: %s' % task_info_message)
 
+        failed_tasks = []
         for runner in task_runners:
             # Mark runner as running
             runner.scan_task.start()
@@ -111,9 +113,13 @@ class ScanJobRunner(Process):
 
             if task_status != ScanTask.COMPLETED:
                 # Task did not complete successfully so save job status as fail
-                error_message = 'One of more tasks did not complete.  '\
-                    'See tasks for details.'
-                self.scan_job.fail(error_message)
+                failed_tasks.append(runner.scan_task)
+
+        if bool(failed_tasks):
+            failed_task_ids = ', '.join([str(task.id)
+                                         for task in failed_tasks])
+            error_message = 'The following tasks failed: %s' % failed_task_ids
+            self.scan_job.fail(error_message)
 
         if self.scan_job.scan_type == ScanTask.SCAN_TYPE_INSPECT:
             self._send_facts_to_engine()

--- a/quipucords/scanner/manager.py
+++ b/quipucords/scanner/manager.py
@@ -62,7 +62,8 @@ class Manager(Thread):
         """Start to excute scans in the queue."""
         if len(self.scan_queue) > 0:  # pylint: disable=C1801
             self.current_task = self.scan_queue.pop()
-            if self.current_task.scan_job.status == ScanTask.PENDING:
+            if self.current_task.scan_job.status in \
+                    [ScanTask.PENDING, ScanTask.RUNNING]:
                 logger.debug('Scan manager running %s.', self.current_task)
                 self.current_task.start()
                 self.current_task.join()

